### DISCUSSION TITLE
Microsoft.PowerShell.SDK incompatible with .NET Core since 7.1?

### DISCUSSION THREAD

**Discussion by DmitryMashkov-DM** [PowerShell/PowerShell#17846] [General]
### Prerequisites

- [X] Write a descriptive title.
- [X] Make sure you are able to repro it on the [latest released version](https://github.com/PowerShell/PowerShell/releases)
- [X] Search the existing issues.
- [X] Refer to the [FAQ](https://github.com/PowerShell/PowerShell/blob/master/docs/FAQ.md).
- [X] Refer to [Differences between Windows PowerShell 5.1 and PowerShell](https://docs.microsoft.com/powershell/scripting/whats-new/differences-from-windows-powershell).

### Steps to reproduce

Microsoft.PowerShell.SDK 7.0.11:
Dependencies:
.NETCoreApp,Version=3.1
Microsoft.PowerShell.SDK 7.1.0+:
Dependencies:
net5.0

When building an app targeting .NET Core, PowerShell namespaces like System.Management.Automation are not available. Is it really impossible to use recent versions of SDK for .NET Core?

### Expected behavior

```console
Latest SDKs should target .NET Core
```

### Actual behavior

```console
.NET Core is not listed.
```

### Environment data

```powershell
Windows 11, Visual Studio 2022.
```

**Comment by fMichaleczek** [comment]
@DmitryMashkov-DM pwsh.exe is produced with the PowerShell.SDK :) Please post a repository where I can see your project (minimal if possible). Because I think you miss something.

**Reply by DmitryMashkov-DM** [reply]
Yes, this must be something I miss because I just can't even write a single line with Core, only with .NET 6.
[Choosing the right PowerShell NuGet package for your .NET project](https://docs.microsoft.com/en-us/powershell/scripting/dev-cross-plat/choosing-the-right-nuget-package?view=powershell-7.2) says:

> A given Microsoft.Powershell.SDK version contains the concrete implementation of the same version of the PowerShell application; version 7.0 contains the implementation of PowerShell 7.0 and running commands or scripts with it will largely behave like running them in PowerShell 7.0.

> Targeting Microsoft.Powershell.SDK from a .NET application allows you to integrate with all of PowerShell's implementation assemblies, such as System.Management.Automation, Microsoft.PowerShell.Management, and other module assemblies.

In fact:  [Microsoft.PowerShell.SDK](https://www.nuget.org/packages/Microsoft.PowerShell.SDK/#supportedframeworks-body-tab) lists only:
net6.0 net6.0-android net6.0-ios net6.0-maccatalyst net6.0-macos net6.0-tvos net6.0-windows
as supported.

The last version that listed .NET Core as supported was 7.0.12.

**Reply by jborean93** [reply]
PowerShell 7.1 is based on dotnet 5.0, 7.2 is based on dotnet 6.0 as per https://docs.microsoft.com/en-us/powershell/scripting/install/powershell-support-lifecycle?view=powershell-7.2#release-history. If you want to target netcoreapp3.0 then you need to reference System.Management.Automation or Microsoft.PowerShell.SDK at the 7.0 versions.

**Reply by bergmeister** [reply]
yes, or PowerShellStandard. YMMV vary based on the APIs that you call. In the worst case you need to do conditional compilation for different ps versions.

**Reply by SeeminglyScience** [reply]
@DmitryMashkov-DM Probably worth mentioning that .NET Core was rebranded to just .NET as of 5.0

So `.NET 5.0` is a continuation of `.NET Core 3.x` *not* `.NET Framework 4.x`

Super confusing for the transition period, but hopefully more clear in the long term.
